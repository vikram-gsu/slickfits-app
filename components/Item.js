import React from 'react';
import Link from 'next/link';

import ItemStyle from './styles/ItemStyles';
import Title from './styles/Title';
import PriceTag from './styles/PriceTag';
import formatMoney from '../lib/formatMoney';

const Item = ({ id, title, description, price, image }) => {
  return (
    <ItemStyle>
      {image && <img src={image} alt={title} />}
      <Title>
        <Link
          href={{
            pathname: '/item',
            query: { id }
          }}
        >
          <a>{title}</a>
        </Link>
      </Title>
      <PriceTag>{formatMoney(price)}</PriceTag>
      <p>{description}</p>
      <div className="buttonList">
        <Link
          href={{
            pathname: 'update',
            query: { id }
          }}
        >
          <a>Edit</a>
        </Link>
        <a>Add To Cart</a>
        <a>Delete This Item</a>
      </div>
    </ItemStyle>
  );
};

export default Item;
