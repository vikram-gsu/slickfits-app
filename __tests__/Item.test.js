import {shallow} from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import Item from '../components/Item';

const fakeItem = {
    id: '12321jlk', 
    title: 'Adidas shoes', 
    description: 'Yellow shoes with three stripes!', 
    price: 3450, 
    image: "adidas.jpg", 
    largeImage: "large_adidas.jpg" 
}
describe('<Item />', () => {
    it('renders and displays properly', () => {
        const wrapper = shallow(<Item {...fakeItem} />);

        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders price as expected', () => {
        const wrapper = shallow(<Item {...fakeItem} />);
        
        const PriceTag = wrapper.find('PriceTag');
        expect(PriceTag.text()).toEqual('$34.50');
    })
    
    it('renders image as expected', () => {
        const wrapper = shallow(<Item {...fakeItem} />);
        const img = wrapper.find('img');
        const imgProps = img.props();
        expect(imgProps.src).toEqual(fakeItem.image);
        expect(imgProps.alt).toEqual(fakeItem.title);
    })
})
