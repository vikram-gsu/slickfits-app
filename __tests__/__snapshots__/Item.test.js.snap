// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and displays properly 1`] = `
<ItemStyles__Item>
  <img
    alt="Adidas shoes"
    src="adidas.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "12321jlk",
          },
        }
      }
      prefetch={true}
    >
      <a>
        Adidas shoes
      </a>
    </Link>
  </Title>
  <PriceTag>
    $34.50
  </PriceTag>
  <p>
    Yellow shoes with three stripes!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "12321jlk",
          },
        }
      }
      prefetch={true}
    >
      <a>
        Edit
      </a>
    </Link>
    <a>
      Add To Cart
    </a>
    <DeleteItem
      id="12321jlk"
    >
      Delete This Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
